service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/clone/azure/database/RestoreAzureDatabaseStep.java

```diff
@@ -0,0 +1,129 @@
+import bio.terra.workspace.service.workspace.WorkspaceService;
+import bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys;
+import bio.terra.workspace.service.workspace.model.AzureCloudContext;
+import java.util.UUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class RestoreAzureDatabaseStep implements Step {
+  private static final Logger logger = LoggerFactory.getLogger(RestoreAzureDatabaseStep.class);
+
+  private final LandingZoneApiDispatch landingZoneApiDispatch;
+  private final SamService samService;
+  private final WorkspaceService workspaceService;
+  private final AzureStorageAccessService azureStorageAccessService;
+  private final AzureDatabaseUtilsRunner azureDatabaseUtilsRunner;
+
+  public RestoreAzureDatabaseStep(
+      LandingZoneApiDispatch landingZoneApiDispatch,
+      SamService samService,
+      WorkspaceService workspaceService,
+      AzureStorageAccessService azureStorageAccessService,
+      AzureDatabaseUtilsRunner azureDatabaseUtilsRunner) {
+
+    this.landingZoneApiDispatch = landingZoneApiDispatch;
+    this.samService = samService;
+    this.workspaceService = workspaceService;
+    this.azureStorageAccessService = azureStorageAccessService;
+    this.azureDatabaseUtilsRunner = azureDatabaseUtilsRunner;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext context) throws InterruptedException, RetryException {
+
+    var inputParameters = context.getInputParameters();
+    var workingMap = context.getWorkingMap();
+
+    ControlledAzureDatabaseResource destinationDatabase =
+        getRequired(
+            workingMap,
+            WorkspaceFlightMapKeys.ControlledResourceKeys.CLONED_RESOURCE_DEFINITION,
+            ControlledAzureDatabaseResource.class);
+
+    var destinationWorkspaceId =
+        getRequired(
+            inputParameters,
+            WorkspaceFlightMapKeys.ControlledResourceKeys.DESTINATION_WORKSPACE_ID,
+            UUID.class);
+    var userRequest =
+        getRequired(
+            inputParameters,
+            JobMapKeys.AUTH_USER_INFO.getKeyName(),
+            AuthenticatedUserRequest.class);
+
+    var destinationContainer =
+        getRequired(
+            workingMap,
+            WorkspaceFlightMapKeys.ControlledResourceKeys.AZURE_STORAGE_CONTAINER,
+            ControlledAzureStorageContainerResource.class);
+
+    var sasToken =
+        azureStorageAccessService.createAzureStorageContainerSasToken(
+            destinationWorkspaceId, destinationContainer, userRequest, null, null, "rcwl");
+    var blobContainerUrlAuthenticated = sasToken.sasUrl();
+
+    var blobFileName =
+        workingMap.get(
+            WorkspaceFlightMapKeys.ControlledResourceKeys.CLONE_DB_DUMPFILE, String.class);
+
+    // Query LZ for the postgres server info
+    var bearerToken = new BearerToken(samService.getWsmServiceAccountToken());
```

Comment:
sonarcloud complains about the fact that this method is annotated with `@VisibleForTesting`

Reply:
WSM team says: we don't care